cpu/nrf51822/radio/nrfmin/nrfmin.c

```diff
@@ -0,0 +1,670 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_nrf51822_nrfmin
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the nrfmin NRF51822 minimal radio driver
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "thread.h"
+#include "sched.h"
+#include "kernel.h"
+#include "periph_conf.h"
+#include "periph/cpuid.h"
+#include "nrfmin.h"
+#include "net/ng_netbase.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief   Driver specific device configuration
+ * @{
+ */
+#define CONF_MODE               RADIO_MODE_MODE_Nrf_2Mbit
+#define CONF_PAYLOAD_LEN        (250U)
+#define CONF_LEN                (8U)
+#define CONF_S0                 (0U)
+#define CONF_S1                 (0U)
+#define CONF_STATLEN            (0U)
+#define CONF_BASE_ADDR_LEN      (4U)
+#define CONF_ENDIAN             RADIO_PCNF1_ENDIAN_Big
+#define CONF_WHITENING          RADIO_PCNF1_WHITEEN_Disabled
+#define CONF_CRC_LEN            (2U)
+#define CONF_CRC_POLY           (0x11021)
+#define CONF_CRC_INIT           (0xf0f0f0)
+/** @} */
+
+/**
+ * @brief   Driver specific address configuration
+ * @{
+ */
+#define CONF_ADDR_PREFIX0       (0xE7E7E7E7)
+#define CONF_ADDR_BCAST         (0xffff)
+/** @} */
+
+/**
+ * @brief   Driver specific (interrupt) events (not all of them used currently)
+ * @{
+ */
+#define ISR_EVENT_RX_START      (0x0001)
+#define ISR_EVENT_RX_DONE       (0x0002)
+#define ISR_EVENT_TX_START      (0x0004)
+#define ISR_EVENT_TX_DONE       (0x0008)
+#define ISR_EVENT_WRONG_CHKSUM  (0x0010)
+/** @} */
+
+/**
+ * @brief   Possible internal device states
+ */
+typedef enum {
+    STATE_OFF,              /**< device is powered off */
+    STATE_IDLE,             /**< device is in idle mode */
+    STATE_RX,               /**< device is in receive mode */
+    STATE_TX,               /**< device is transmitting data */
+} state_t;
+
+/**
+ * @brief   In-memory structure of a nrfmin radio packet
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t length;                     /**< packet length */
+    uint8_t src_addr[2];                /**< source address of the packet */
```

Comment:
This I will not change. The semantics are clear and a 2 is faster to read/grasp then another define...